app/forms/candidate_interface/contact_details_form.rb

```diff
@@ -35,6 +39,15 @@ def self.build_from_application(application_form)
       )
     end
 
+    def sanitize_phone_number
+      self.raw_phone_number = phone_number
+      self.phone_number = phone_number.gsub(/[\s()\-+]/, '') unless phone_number.nil?
+    end
+
+    def reset_phone_number
+      self.phone_number = raw_phone_number
+    end
```

Comment:
This feels a bit unexpected to me. Could we move this logic to the `PhoneNumberValidator` class instead, and move the length constraints there too?

Something like this maybe?

```ruby
class ModelName
  validates :phone_number, phone_number: true
end

class PhoneNumberValidator < ActiveModel::EachValidator
  def validate_each(record, attribute, value)
    if value.blank?
      record.errors.add(attribute, :blank)
    elsif invalid_phone_number_format?(value)
      record.errors.add(attribute, :invalid)
    elsif not_enough_digits?
      record.errors.add(attribute, :too_short)
    elsif too_many_digits?
      record.errors.add(attribute, :too_long)
    end
  end
  
  private
  
  def invalid_phone_number_format?(value)
    value.match?(/[^ext()+. 0-9]/)
    value.match?(/[^-()+. 0-9]/)
  end
  
  def not_enough_digits?
    value.gsub(/[^0-9]/, '').length < 8
  end

  def too_many_digits?
    value.gsub(/[^0-9]/, '').length > 15
  end
   
end
```

That way we'd also avoid displaying two messages (one about length and one about format) for the same field – better to only show one at a time.

Reply:
Thanks Frankie, we fixed this in 698eaea6a86443db9e441f48f9acee85e770ee5b